### PR TITLE
🏁 Cancel flag capture and better logging.

### LINKED ISSUE #2215: [Feature Request]: Hint the location of nearby enemies when attempting to capture a flag.
### Related problem

Sometimes outposts can be recapped by a single soldier who is hiding within the capture radius. Searching an area just isn't fun for humans. However, flags can be taken if the rebels outnumber the nearby AI.

### Proposed feature

Block the flag from being captured, but instead, hint to the nearby location of Enemys.
Carinal directions can be used. Like N, NW, ENE ect.

### Alternative solutions

Play a coughing noise on the enemy that is loud enough to hear from the flag pole. Maybe this is both better for immersion and easier to triangulate when working with a squad.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CalebSerafin**
## What type of PR is this.
* [x] Bug-fix
* Change
* [x] Enhancement

### What have you changed and why?
* While capturing an outpost, you can abort the action and immediately return to weapon up.
* You can see the remaining outpost flag cooldown when trying to spam the flag.

### Please specify which Issue this PR Resolves.
none

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
* Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
1. While capturing an outpost, you can abort the action and immediately return to weapon up.
2. You can see the remaining flag cooldown when trying to spam the flag.

### CODE REVIEW

**Review by Lazejun — COMMENTED** [review]
Played around with it, 
Skipped Step 1,
2: Yes seems to work pretty well, i personally find it silly but that's my Opinon, its still useful for most Players as its Visual and audible.

In Theory one Person can spam it and one person can check for Smoke/Coughs and just destroy that Gridref,
It's also Funny as it work on approaching Vehicles etc.



3: Yes you are not stuck in an Animation anymore :D .
4: Yes and you even Included ms.









we might have to Update the Hint as there is nothing to Check on the Map

**Comment by CalebSerafin** [comment]
Hmm, yeah. That lack of enemy markers may be the game difficulty.
We did discuss the silliness of the magical flag causing enemies to reveal themselves.

The smoke and cough will be replaced by casual voice lines that play regardless (However, it depends on the user's having audio if their difficultly blocks enemy markers.). But the other changes will remain.

File: A3A/addons/core/functions/Base/fn_mrkWIN.sqf [hunk 1]
```diff
@@ -45,21 +69,49 @@ private _enemyValue = 0;
...
 } forEach (allUnits inAreaArray [_markerPos, _capRadius, _capRadius]);
 
-ServerDebug_2("Rebel value %1, enemy value %2", _rebelValue, _enemyValue);
 
 if (_enemyValue > 2*_rebelValue) exitWith
 {
-    ServerInfo_1("Flag capture by %1 abandoned due to outnumbering.", str _playerX);
-    ["Capture", "The enemy still outnumber us, check the map and clear the rest of the area."] call A3A_fnc_customHint;
+    ServerInfo_4("Outpost at %1 (%2): Flag capture cancelled due to enemy value (%3) greater than 2*rebel value (%4)", _outpostGridSquare, _markerX, _enemyValue, _rebelValue);
+    if (playerMarkersEnabled) then {
```
Comment (Lazejun) [inline A3A/addons/core/functions/Base/fn_mrkWIN.sqf:80]: The Message will be wrong regardless of the Parameter as it has nothing todo with Players using different Difficulty settings.

https://community.bistudio.com/wiki/Arma_3:_Difficulty_Settings

One solution would be to check for the actual setting

```sqf
difficultyOption "mapContentEnemy" //1 is on
```

https://community.bistudio.com/wiki/difficultyOption

Comment (CalebSerafin) [reply A3A/addons/core/functions/Base/fn_mrkWIN.sqf]: Thank you. Will implement.

**Review by Lazejun — APPROVED** [review]
Abort works quickly,
Hint message without MapContentEnemy works,
Hint message with MapContentEnemy works,
Anti Spam cooldown works.

**Pull request merged**